api/v1alpha1/envoyproxy_types.go

```diff
@@ -115,6 +120,25 @@ type EnvoyProxyProvider struct {
 	Kubernetes *EnvoyProxyKubernetesProvider `json:"kubernetes,omitempty"`
 }
 
+// ShutdownConfig defines configuration for graceful envoy shutdown process.
+type ShutdownConfig struct {
+	// DrainTimeout defines the graceful shutdown timeout. This should be less than the pod's terminationGracePeriodSeconds.
```

Comment:
10min seems like a long time for a default value
I do understand that this will be lesser in case active connections fall to 0, but 10min seems long

Reply:
I don't think 10 minutes is all that high for an edge proxy, and the docs for the drain process seem to agree:

> In edge scenarios, it might be desirable to have a very long drain time. In service to service scenarios, it might be possible to make the drain and shutdown time much shorter (e.g., 60s/90s).

ref: https://www.envoyproxy.io/docs/envoy/latest/operations/cli#cmdoption-drain-time-s

Case in point, some non-Envoy balancers in front on some of our k8s Ingresses have a 20 minute drain period for HTTP requests. Streaming and TCP connections (which can't be signaled to gracefully close) can also easily run up the clock.

Reply:
thanks for sharing that link, default value of `10m` comes from envoy itself, so sgtm